common/src/main/java/com/genexus/ICacheService.java

```diff
@@ -1,12 +1,15 @@
 package com.genexus;
 
+import java.util.List;
+
 public interface ICacheService
 {
 	boolean containtsKey(String cacheid, String key);
 	<T> T get(String cacheid, String key, Class<T> type);
+	<T> List<T> getAll(String cacheid, String[] keys, Class<T> type);
 	<T> void set(String cacheid, String key, T value);
 	<T> void set(String cacheid, String key, T value, int duration);
-
+	<T> void setAll(String cacheid, String[] keys, T value);
```

Comment:
Two comments: 
1. (minor) As the getAll returns a List of Values, the SetAll could receive different values for each key. 
It's ok, I understand is not needed right now. 

2. SetAll does not receive Expiration timeout. So, this keys will live forever. 
Just to be sure. Before this change, we were setting this value without expiration? I'm a little bit reticent to push data to Cache without any expiration. 
- InProcess Cache, when WebServer (or Tomcat) restarts, the Cache starts clean so all the Data will be Fresh.
- In Distributed Cache, when WebServer (or Tomcat) restarts, we will have the same behaviour?

Reply:
1. Thats the reason there is no SetAll with differente values, because it is not neeeded. In the case of SD cache It would add some overhead to create a list of the same value for all the keys just to honor the interface.

2. Items in SD cache don´t need an expiraton date because each item in SD cache is a pair [key:table, value:LastModifiedDateOfTable], so it works more like a DB instead of a cache, that caches the LastModifiedDate of each table (and that information doesn´t expire, it is just refreshed -set- when a table changes).

In Distributed Cache, when WebServer (or Tomcat) restarts, we will have the same behaviour? => yes, we will. Because StartupDate of app is also considered as a condition when checking the LastModifiedDates of services tables.  A service returns [403 not modified] only when LastModifiedDate of all input tables and app.StartupDate are before if-modified-Since header sent by SD device.

Reply:
1. Changed SetAll to improve reusability.